velox/exec/Driver.cpp

```diff
@@ -1090,6 +981,72 @@ folly::dynamic Driver::toJson() const {
   return obj;
 }
 
+template <typename Func>
+auto Driver::withDeltaCpuWallTimer(
+    Operator* op,
+    TimingMemberPtr timingMember,
+    Func&& timedFunction) -> decltype(timedFunction()) {
+  // If 'trackOperatorCpuUsage_' is true, create and initialize the timer object
+  // to track cpu and wall time of the timedFunction.
+  if (!trackOperatorCpuUsage_) {
+    return timedFunction();
+  }
+
+  // The delta CpuWallTiming object would be recorded to the corresponding
+  // timingMember upon destruction of the timer when withDeltaCpuWallTimer
+  // ends. The timer is created on the stack to avoid heap allocation
+  auto f = [op, timingMember, this](const CpuWallTiming& elapsedTime) {
+    auto elapsedSelfTime = processLazyTiming(*op, elapsedTime);
```

Comment:
We can pass time tracking as a lambda function to withDeltaCpuWallTimer? Thanks!

Reply:
> We can pass time tracking as a lambda function to withDeltaCpuWallTimer? Thanks!
@xiaoxmeng Sorry I don't quite get what you meant, could you please explain a bit? Thanks

Reply:
Ok. I thought to pass the time tracking logic as a lambda function but you try to dedup in this PR.